sdk/core/logger/README.md

```diff
@@ -0,0 +1,90 @@
+# Azure Logger library for JS
+
+The `@azure/logger` package can be used to enable logging in the Azure SDKs for JavaScript.
+
+Logging can be enabled for the Azure SDK in the following ways:
+
+- Setting the AZURE_LOG_LEVEL environment variable
+- Calling setLogLevel imported from "@azure/logger"
+- Calling enable() on specific loggers
+- Using the `DEBUG` environment variable.
+
+Note that AZURE_LOG_LEVEL, if set, takes precedence over DEBUG. Only use DEBUG without specifying AZURE_LOG_LEVEL or calling setLogLevel.
+
+## Getting started
+
+### Installation
+
+Install this libray using npm as follows
+
+```
+npm install @azure/logger
+```
+
+## Key Concepts
+
+The `@azure/logger` package supports the following log levels
+specified in order of most verbose to least verbose:
+
+- verbose
+- info
+- warning
+- error
+
+When setting a log level, either programmitcally or via the `AZURE_LOG_LEVEL` environment variable,
+any logs that are written using a log level equal to or less than the one you choose
+will be emitted.
+
+For example, setting the log level to `warning` will cause all logs that have the log
+level `warning` or `error` to be emitted.
+
+## Examples
+
+### Example 1 - basic usage
+
+```js
+const { EventHubClient } = require('@azure/event-hubs');
+
+const logger = require('@azure/logger');
+logger.setLogLevel('info');
+
+// operations will now emit info, warning, and error logs
+const client = new EventHubClient(/* params */);
+client.getPartitionIds()
+  .then(ids => { /* do work */ })
+  .catch(e => { /* do work */ });
+});
+```
+
+### Example 2 - redirect log ouput
+
+```js
+const { AzureLogger, setLogLevel } = require('@azure/logger');
+
+setLogLevel("verbose");
+
+// override logging to output to console.log
+AzureLogger.log = (...args) => {
+  console.log(...args);
```

Comment:
Isnt console.log the default?
If so, and we want to talk about overriding here, then should we use a different example?

Reply:
Console.error is the default. Could add a comment above saying like // use console.log instead of console.error

Reply:
Comment added